bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -62,63 +59,73 @@ public void unsetScriptExtensionManager(ScriptExtensionManager scriptExtensionMa
     }
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    public void addScriptEngineFactory(ScriptEngineFactory provider) {
-        this.scriptEngineFactories.add(provider);
-
-        for (String language : provider.getLanguages()) {
-            this.supportedLanguages.put(language, provider);
+    public void addScriptEngineFactory(ScriptEngineFactory engineFactory) {
+        engineFactory.logFactoryDetails();
+        for (String scriptType : engineFactory.getScriptTypes()) {
+            if (isCustomFactory(engineFactory) == true) {
+                this.customSupport.put(scriptType, engineFactory);
+            } else {
+                this.genericSupport.put(scriptType, engineFactory);
+            }
         }
+        logger.trace("Added {}", engineFactory.getClass().getSimpleName());
     }
 
-    public void removeScriptEngineFactory(ScriptEngineFactory provider) {
-        this.scriptEngineFactories.remove(provider);
-
-        for (String language : provider.getLanguages()) {
-            this.supportedLanguages.remove(language, provider);
+    public void removeScriptEngineFactory(ScriptEngineFactory engineFactory) {
+        this.scriptEngineFactories.remove(engineFactory);
+        for (String scriptType : engineFactory.getScriptTypes()) {
+            if (isCustomFactory(engineFactory) == true) {
+                this.customSupport.remove(scriptType, engineFactory);
+            } else {
+                this.genericSupport.remove(scriptType, engineFactory);
+            }
         }
+        logger.trace("Removed {}", engineFactory.getClass().getSimpleName());
     }
 
-    @Override
-    public boolean isSupported(String fileExtension) {
-        return findEngineFactory(fileExtension) != null;
+    /**
+     * This method is used to determine is a given {@link ScriptEngineFactory} is generic or customized.
+     *
+     * @param engineFactory {@link ScriptEngineFactory}
+     * @return boolean
```

Comment:
```
@return true if the factory is a custom one otherwise true
```

IMHO it does not bring any value to state which data type is returned.

Reply:
Done.